src/posix/fs.cpp

```diff
@@ -1296,33 +1232,20 @@ bool PosixFileSystemAccess::rmdirlocal(LocalPath& name)
 
 bool PosixFileSystemAccess::mkdirlocal(LocalPath& name, bool)
 {
-#ifdef USE_IOS
-    string absolutename;
-    if (appbasepath)
-    {
-        if (name->size() && name->at(0) != '/')
-        {
-            absolutename = appbasepath;
-            absolutename.append(*name);
-            name = &absolutename;
-        }
-    }
-#endif
-
     mode_t mode = umask(0);
-    bool r = !mkdir(name.editStringDirect()->c_str(), defaultfolderpermissions);
+    bool r = !mkdir(adjustBasePath(name).c_str(), defaultfolderpermissions);
     umask(mode);
 
     if (!r)
     {
         target_exists = errno == EEXIST;
         if (target_exists)
         {
-            LOG_debug << "Error creating local directory: " << name.editStringDirect()->c_str() << " errno: " << errno;
+            LOG_debug << "Error creating local directory: " << adjustBasePath(name) << " errno: " << errno;
         }
         else
         {
-            LOG_err << "Error creating local directory: " << name.editStringDirect()->c_str() << " errno: " << errno;
+            LOG_err << "Error creating local directory: " << adjustBasePath(name) << " errno: " << errno;
```

Comment:
using a variable will speed things up for ios

Reply:
This applies to other PosixFileSystemAccess methods

Reply:
I've applied your suggestion where I think it makes a difference. Please, have a look and approve the PR, if everything is fine. Thanks